### PR TITLE
REF/BLD: migrate to latest standards (pyproject.toml) and unpin pyqt

### PR DESCRIPTION
**Pull request by klauer**
## Description
* Unpin pyqt, leaving it at 5.x.
* Upgrading by way of https://github.com/pcdshub/pcds-python-migration-tools
* hxrsnd has been migrated to use setuptools-scm, replacing versioneer, as its version-string management tool of choice.
* hxrsnd has been migrated to use the modern ``pyproject.toml``, replacing ``setup.py`` and related files.
* Older language features and syntax found in the repository have been updated to Python 3.9+ standards by way of ``pyupgrade``.
* Sphinx 6.0 is now supported for documentation building. ``docs-versions-menu`` replaces ``doctr-versions-menu`` and ``doctr`` usage for documentation deployment on GitHub Actions.  The deployment key is now no longer required.
* `pyupgrade` was used to update annotations, though there might not have been many here
* `pycln` was used to clean up imports
* Some manual fixes were required to fix up style
* Migrated repository settings to GitHub Actions passing requirement for PR merging.

## Motivation and Context
We're looking forward to Python 3.11 everywhere.

`hutch-python > hxrsnd` dependency means that hutch-python and these libraries currently require `pyqt < 5.15` in our environment. Those versions are not available for Python 3.11. PyDM and our other libraries are now (likely/mostly) compatible with PyQt 5.15, so releasing the pin here is our next step toward that transition.

Relying on outdated CI infrastructure required an update to this repository prior to moving forward.

hxrsnd was not previously on our list of repositories to upgrade, but this seemed to be the path of least resistance for now.

## How Has This Been Tested?
Untested. Relying on CI entirely.

## Where Has This Been Documented?
This PR text.

### CODE REVIEW

**Comment by klauer** [comment]
(How... did Travis run here? Are we mistakenly still paying? I somehow no longer have access to the admin page there...)

**Comment by klauer** [comment]
There are entrypoint scripts that I'm not entirely sure what to do with here... Hmm:

https://pcdshub.github.io/hxrsnd/v0.2.4/scripting.html

**Comment by ZLLentz** [comment]
As far as I can tell from inspecting the xcs hutch python, the scripts.py thing here was copied over into xcs3 next to their beamline file as snd_scripts.py and is used in that context instead.

I am not sure if run_snd is used directly but your adjustments here all are correct if it is.

Otherwise, this hxrsnd module is simply imported during the beamline.py and instantiated as:
```
with safe_load('Split and Delay'):
   from hxrsnd.sndsystem import SplitAndDelay
   from xcs.db import daq, RE
   snd = SplitAndDelay('XCS:SND', name='snd', daq=daq, RE=RE)
```

**Review by ZLLentz — APPROVED** [review]
Looks correct to me

File: hxrsnd/bragg.py [hunk 1]
```diff
@@ -302,7 +302,7 @@ def d_space(ID, hkl):
     ID = check_id(ID)
     h = hkl[0]
     k = hkl[1]
-    l = hkl[2]
+    l_ = hkl[2]
```
Comment (tangkong) [inline hxrsnd/bragg.py:305]: Out of curiosity, what was the problem here?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+file = "README.rst"
+
+[tool.setuptools.dynamic.dependencies]
+file = [ "requirements.txt",]
+
+[tool.setuptools.dynamic.optional-dependencies.test]
+file = "dev-requirements.txt"
+
+[tool.setuptools.dynamic.optional-dependencies.doc]
+file = "docs-requirements.txt"
```
Comment (tangkong) [inline pyproject.toml:43]: Should we add the coverage option here?

```
[tool.pytest.ini_options]
addopts = "--cov=."
```

**Review by tangkong — APPROVED** [review]
LGTM

File: hxrsnd/bragg.py (hunk 1 above)
Comment (klauer) [reply hxrsnd/bragg.py:305]: "Ambiguous variable name" `l` looks like `1`

**Pull request merged**